.github/workflows/matrix.yml

```diff
@@ -239,23 +253,27 @@ jobs:
       with:
         path: ${{ steps.get-vars.outputs.ccache-path }}
         # double-dash after compiler is not a typo, it is to disambiguate between g++-<date> and g++-11-<date> for restore key prefix matching
-        key: ccache-${{ runner.os }}-${{ matrix.compiler }}--${{ steps.get-vars.outputs.date }}
+        key: ccache-${{ github.ref_name }}-${{ runner.os }}-${{ matrix.compiler }}--${{ steps.get-vars.outputs.datetime }}
```

Comment:
This part does not work AFAICT.
If we have a miss on this key, we will emit an (empty) cache object, and if we have another build before a master build refreshes the cache timestamp, we will pull the empty cache.
If the timestamp *is* refreshed we will instead fall back to the master cache.

I *think* what we want is to have the same key for both master and PRs, and mostly rely on PR cache retrieval hitting on the primary key.

Reply:
I changed it to use datetime instead of date, so every build will always be 'missing' on its primary key and falling back to master. Unless a build runs, finishes, uploads to cache, and reschedules another run in the same minute, a PR should never be hitting its own cache.

The only time a PR will run without any cache would be if the master cache were evicted or the cachekey bumped.

Reply:
Ok cool i was too distracted with my preconceived notion, but this is simpler.

Master:
Always misses master-config-time
secondary hit on master-config-
emits cache master-config-time

Pr:
Always misses PR-config-time
secondary hit on master-config-
emits (empty) cache PR-config-time

Reply:
Exactly.